libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -849,7 +849,6 @@ private async Task<bool> QueueFirstMatchAsync(ActionContext actionContext, Dialo
                     { "DialogId", Id },
                     { "Expression", evt.GetExpression().ToString() },
                     { "Kind", $"Microsoft.{evt.GetType().Name}" },
-                    { "Instance", JsonConvert.SerializeObject(evt, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }).ToString() }
```

Comment:
@garypretty What do you suggest here that is not serializing the entire evt? It has a reference to the dialog that could be circular or just large.

Reply:
For other folks - while trying this code with rather large json structures we found that this line was extremely expensive and potentially dangerous as it could lead to large object heap usage and it's simply a lot of potentially unnecessary serialization.

Reply:
I would like @vishwacsena to comment on this.  @vishwacsena has asked for all of the properties / scope for the event to be captured as part of telemetry.  I am not familiar enough with adaptive right now to definitively say what we should / should not capture here.

Reply:
Expression should be sufficient.  and trigger ID, 
The problem with doing the entire object is that you are doing a deep serializing of ALL of the actions for a trigger, including recognizers, generators, EVERYTHING for all steps in the trigger.

---
In reply to: [419414292](https://github.com/microsoft/botbuilder-dotnet/pull/3857#discussion_r419414292) [](ancestors = 419414292)

Reply:
As with below, I think for this PR, removing the Instance property is enough.  We can always make additive changes to the captured telemetry in a future release. I think we're good.